src/transactions/TransactionFrameBase.h

```diff
@@ -94,5 +94,6 @@ class TransactionFrameBase
     getDiagnosticEvents() const = 0;
     virtual int64 declaredSorobanResourceFee() const = 0;
     virtual bool XDRProvidesValidFee() const = 0;
+    virtual uint64_t getSize() const = 0;
```

Comment:
I don't think this needs to be an interface method, you're only using it for the regular tx frame (so it can be just a private method there). Note, that if it's going to be used by the external clients, the function should be renamed to something like `getEnvelopeSizeForSoroban` or something like that  and return the size of the inner envelope (not the whole envelope). But again, I'm not sure if that's necessary at all.

Reply:
I agree this doesn't need to be exposed like this, previously I had added 
`uint64_t getTransactionEnvelopeSize(TransactionEnvelope const& envelope)` to TransactionUtils, but after the flow control accounting simplification, I moved the functionality `TransactionFrameBase::getSize`. Either this needs to be a private function as you said or I need to revert the last couple commits and have 
`TransactionUtils::getTransactionEnvelopeSize`. Depends on what level we want to "centralize" the size calculation, thoughts?

Reply:
Well, I'd say this should be dictated by the users of this. Maybe there is no longer a significant benefit from this. If the only remaining user is tx frame itself (it seems like it is), then this should be just a private helper. I quickly looked through the code and only found the usage in tests (besides tx frame) - we could make the helper public for tests, but that would require a downcast from tx frame base to tx frame, so I don't think it's worth the hassle.

Reply:
Cool, I made `getSize` a private function of `TransactionFrame` and removed it from `TransactionFrameBase`.